NeoML/include/NeoML/TraditionalML/SubwordEncoder.h

```diff
@@ -96,11 +103,18 @@ class NEOML_API ISubwordEncoderWithCache : public ISubwordEncoder {
 	mutable CCache cache;
 };
 
+DECLARE_NEOML_MODEL_NAME( BytePairEncoderModelName, "NeoMLBytePairEncoderModel" )
+
 class NEOML_API IBytePairEncoder : public ISubwordEncoderWithCache {
 public:
 	// Returns encoder flags.
 	virtual bool UseEndOfWordToken() const = 0;
 	virtual bool UseStartOfWordToken() const = 0;
+
+	// Initializes the encoder. Can be safely used only once.
+	virtual void LoadDictionary( const CWordDictionary& tokens, bool useEndOfWordToken, bool useStartOfWordToken ) = 0;

```

Comment:
By public `LoadDictionary` we introduce possibility when the content of `tokens` is inconsistent with values of `useEndOfWordToken` and `useStartOfWordToken`.
The initial idea was to encapsulate build process into Trainer class, so we can guarantee that resulting Encoder is consistent and valid.

Reply:
`LoadDictionary`  can be useful for experiments. For example, we want to try external (sentence-piece trained on another corpus) vocabulary. Looks like we can easily check `tokens` for our special subtokens and raise exception if any exists.